### ISSUE TITLE
Create index request should return the index name

### ISSUE THREAD

**Issue by clintongormley** [elastic/elasticsearch#23044]
The create index request should return the index name, as it is not always obvious which index has been created.

For instance, the following create-index request uses date math in the name:

    PUT <logs-{now}>

You have to either do the date calculation yourself or look at the list of indices to figure out what the index name is.

**Comment by DanielLyons178** [comment]
I am happy to take a look at this as my first PR. I assume the change would go [here](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java#L42-L56) 
Changing 
```
public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
        CreateIndexRequest createIndexRequest = new CreateIndexRequest(request.param("index"));
        if (request.hasContent()) {
            createIndexRequest.source(request.content(), request.getXContentType());
        }
        createIndexRequest.updateAllTypes(request.paramAsBoolean("update_all_types", false));
        createIndexRequest.timeout(request.paramAsTime("timeout", createIndexRequest.timeout()));
        createIndexRequest.masterNodeTimeout(request.paramAsTime("master_timeout", createIndexRequest.masterNodeTimeout()));
        createIndexRequest.waitForActiveShards(ActiveShardCount.parseString(request.param("wait_for_active_shards")));
        return channel -> client.admin().indices().create(createIndexRequest, new AcknowledgedRestListener<CreateIndexResponse>(channel) {
            @Override
            public void addCustomFields(XContentBuilder builder, CreateIndexResponse response) throws IOException {
                response.addCustomFields(builder);
            }
        });
}
```
to
```
    public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
        CreateIndexRequest createIndexRequest = new CreateIndexRequest(request.param("index"));
        if (request.hasContent()) {
            createIndexRequest.source(request.content(), request.getXContentType());
        }
        IndexNameExpressionResolver nameExpressionResolver = new IndexNameExpressionResolver(settings);
        String resolvedName =  nameExpressionResolver.resolveDateMathExpression(createIndexRequest.index());

        createIndexRequest.updateAllTypes(request.paramAsBoolean("update_all_types", false));
        createIndexRequest.timeout(request.paramAsTime("timeout", createIndexRequest.timeout()));
        createIndexRequest.masterNodeTimeout(request.paramAsTime("master_timeout", createIndexRequest.masterNodeTimeout()));
        createIndexRequest.waitForActiveShards(ActiveShardCount.parseString(request.param("wait_for_active_shards")));
        return channel -> client.admin().indices().create(createIndexRequest, new AcknowledgedRestListener<CreateIndexResponse>(channel) {
            @Override
            public void addCustomFields(XContentBuilder builder, CreateIndexResponse response) throws IOException {
                builder.field("index-name", resolvedName);
                response.addCustomFields(builder);
            }
        });
    }
```
If this is what you had in mind @clintongormley I can submit a PR now

**Comment by jasontedor** [comment]
Thanks for looking @DanielLyons178, but it should be part of the create index response (where it would be automatically rendered on the REST layer), not only the REST layer.

**Comment by jasontedor** [comment]
Also, changing create index response is going to have backwards compatibility concerns.

**Comment by DanielLyons178** [comment]
Thanks for the feedback @jasontedor , after reviewing what you said I made these changes [TransportCreateIndexAction.java ](https://gist.github.com/DanielLyons178/19a27860bd81532c5757130796627e1d) and [CreateIndexResponse.java](https://gist.github.com/DanielLyons178/8f6e5b74738d4a7622d3164a0f8c6abf) and this gives the same results.

**Comment by jasontedor** [comment]
The index name needs to be serialized in the response. This is where backwards compatibility aspects come into play. We will need tests too. It would be easier to communicate these concerns if you opened a PR if you're interested in doing that.

**Comment by s1monw** [comment]
> The index name needs to be serialized in the response. This is where backwards compatibility aspects come into play. We will need tests too. It would be easier to communicate these concerns if you opened a PR if you're interested in doing that.

++ @DanielLyons178 there is no such thing as a perfect first PR. Once we have code we can fix it... we are more than happy to work with you and iterate. I am glad you are picking it up. Don't be afraid we are here to help.

**Comment by allanmc** [comment]
A quick thought, but i guess this need is not isolated to CreateIndexResponse? Wouldn't the same needs apply to every other request that allowed index name expressions in the request? Others like ShrinkResponse, OpenIndexResponse, CloseIndexResponse, DeleteIndexResponse - all would benefit from responding with the actual index(es) it resolved the expression to.
Maybe an idea would be using a common abstract ActionResponse class for those and a RestBuilderListener to match it.

**Comment by kunal642** [comment]
if this bug isn't fixed, i would like to work on it.
Thanks

**Comment by olcbean** [comment]
I have a quick question regarding the workflow that is / should be followed (I guess this is not the most appropriate place to ask but the dev discussion board has been deleted few months back...)

However, here is my question:

@DanielLyons178 was the first to say that he would like to work on the issue and according to the https://github.com/elastic/elasticsearch/pull/23701#issuecomment-288469579 it seems that this made the issue a kind of `assigned` to him. This make sense as there is no need for numerous PRs at the end. However the issue is still left with the `adoptme` label. My understanding would be that any issue with `adoptme` label is fair game, but it appears not to be the case. Could you please advise, so that I do not violate any project agreements.

**Comment by clintongormley** [comment]
@olcbean people frequently volunteer but then don't end up completing, which is why we leave the `adoptme` label in place.  in this case, however, @DanielLyons178 is actively working on a PR here https://github.com/elastic/elasticsearch/pull/23061

**Comment by olcbean** [comment]
@clintongormley thank you for the quick response and swift actions :) 
I saw that @DanielLyons178 is currently working on the issue and that's why I picked this one to ask my question ;)

So just to confirm : In general an issue with `adoptme` label is a fair game until somebody volunteers to work on it. At this point the `adoptme` label should be ignored.
As I am looking to get my feet wet, may I look into issues which still do no have the `adoptme` label, are not explicitly assigned and nobody volunteered for? Or should I wait for a member to explicitly add the `adoptme` label?
